cirq/ops/common_gates.py

```diff
@@ -322,6 +334,43 @@ def _act_on_(self, args: Any):
                                                         tableau.xs[:, q].copy())
             return True
 
+        if isinstance(args, clifford.ActOnStabilizerCHFormArgs):
+            if protocols.is_parameterized(self) or self.exponent % 0.5 != 0:
+                return NotImplemented
+            effective_exponent = self._exponent % 2
+            state = args.state
+            if effective_exponent == 0.5:
+                if ZPowGate()._act_on_(args) is True:
+                    if H._act_on_(args) is True:
+                        state.omega *= (1 + 1j) / (2**0.5)  # type: ignore
+                        return True
+                    # Only one of the delegated _act_on_ succeeded, so the
+                    # state might have been corrupted, return False instead of
+                    # NotImplemented.
+                    return False  # coverage: ignore
+            elif effective_exponent == 1:
+                if ZPowGate()._act_on_(args) is True:
+                    if H._act_on_(args) is True:
+                        if ZPowGate()._act_on_(args) is True:
+                            if H._act_on_(args) is True:
+                                state.omega *= 1j  # type: ignore
+                                return True
+                    # At least one, but not all, of the delegated _act_on_
+                    # succeeded, so the state might have been corrupted, return
+                    # False instead of NotImplemented.
+                    return False  # coverage: ignore
+            elif effective_exponent == 1.5:
+                if H._act_on_(args) is True:
+                    if ZPowGate()._act_on_(args) is True:
+                        state.omega *= (1 - 1j) / (2**0.5)  # type: ignore
+                        return True
+                    # Only one of the delegated _act_on_ succeeded, so the
+                    # state might have been corrupted, return False instead of
+                    # NotImplemented.
+                    return False  # coverage: ignore
```

Comment:
Does the global protocol method recognize these False results? Won't this cause a misleading error about the gate class returning an invalid result, instead of an error about not supporting the operation?

Reply:
Agreed, I was trying to have it trigger that error but asserting as you mentioned above seems like a better way since these gates must have `_act_on_` implemented. So failing right there would be better both design and debugging wise.

Thanks!